src/porepy/models/fluid_mass_balance.py

```diff
@@ -534,9 +503,15 @@ def create_variables(self) -> None:
             tags={"si_units": f"m^{self.nd} * Pa"},
         )
 
-    def pressure(self, subdomains: list[pp.Grid]) -> pp.ad.MixedDimensionalVariable:
-        p = self.equation_system.md_variable(self.pressure_variable, subdomains)
-        return p
+    def pressure(
+        self, grids: list[pp.Grid] | list[pp.BoundaryGrid]
+    ) -> pp.ad.MixedDimensionalVariable:
+        if len(grids) > 0 and isinstance(grids[0], pp.BoundaryGrid):
+            return pp.ad.TimeDependentDenseArray(
+                name=self.bc_data_pressure_key, domains=grids
+            )
+
+        return self.equation_system.md_variable(self.pressure_variable, grids)
```

Comment:
I think we want (better method name needed!):
    
    return self.get_dynamic_quantity(self.pressure_name, grids)

    def get_dynamic_quantity(name, grids) -> Operator:
        if len(grids) > 0 and isinstance(grids[0], pp.BoundaryGrid):
            return pp.ad.TimeDependentDenseArray(
                name=name, domains=grids
            )

        return self.equation_system.md_variable(name, grids)

This can be reused for temperature and displacement

Reply:
Looks good to me. Where would you locate this method? We don't have an abstract BoundaryConditionMixin, it would fit there perfectly. Maybe VariableMixin?

Reply:
Or rather just make that abstract class for BCs?

Reply:
Closing after discussion in person: added an abstract class for BCs.